### ISSUE TITLE
Controller/app versioning

### ISSUE THREAD

**Issue by alfonsogarciacaro** [SaturnFramework/Saturn#23]
As far as I know, GraphQL tries to solve the following problems:

1. Give more power to the client to build queries to the server
2. Provide a data schema that clients can inspect with appropriate tooling
3. Allow different old client versions (particularly mobile apps which are not always up-to-date) to interact with the server

Point 1 is a bit controversial as some think it's not a good idea to give clients the power to query anything in your database. Point 2 is not that necessary if you use a typed language in both ends and share the domain model. But I'm interested in point 3, as with server frequent updates the ability to respond gracefully to out-of-date clients is important. Not only for mobile apps but also in SPAs that can remain outdated until the user refreshes for a long time.

Forcing devs to give a version number can lead them through the pit of success and make it simpler to update the server while keeping different versions of a controller alive during a transition period. This is one of the features I like the most in [IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IDBOpenDBRequest/onupgradeneeded) even if it looks cumbersome at the beginning.

The version number could be in the URL or in a header if we want to hide it, and the framework would take care of calling the appropriate controller. It would be easy to create a helper for Fable so it automatically adds the version to the requests too.

**Comment by Krzysztof-Cieslak** [comment]
I love this idea. I'm not sure if enforcing all the time is a good idea but we can add `version` keyword to `controller` CE that would add version checking. I think I would go with custom header... or maybe with `Accept` header including version. I believe that URL shouldn't change since in RESTful approach it still describes the same resource. (Troy Hunt has written blog post about versioning - https://www.troyhunt.com/your-api-versioning-is-wrong-which-is/ )

It could look like this:
```
let userControllerV1 = controller {
  version 1 //sets version of the controller

  index ActionV1.index
  create ActionV1.create
  ...
}

let userControllerV2 = controller {
  version 2 //sets version of the controller

  index ActionV2.index
  create ActionV2.create
  ...
}

...

let router = scope {
  forward "/users" userControllerV1 
  forward "/users" userControllerV2 
}
```

Thanks a lot for this suggestion, this is exactly the stuff I want to include in Saturn!

**Comment by Krzysztof-Cieslak** [comment]
Implemented in 5274e2382a360f201a4fec51ba843fcf4ffc0ae3
